src/main/java/com/ecore/roles/service/impl/MembershipsServiceImpl.java

```diff
@@ -0,0 +1,55 @@
+package com.ecore.roles.service.impl;
+
+import com.ecore.roles.exception.InvalidArgumentException;
+import com.ecore.roles.exception.ResourceExistsException;
+import com.ecore.roles.exception.ResourceNotFoundException;
+import com.ecore.roles.model.Membership;
+import com.ecore.roles.model.Role;
+import com.ecore.roles.repository.MembershipRepository;
+import com.ecore.roles.repository.RoleRepository;
+import com.ecore.roles.service.MembershipsService;
+import lombok.NonNull;
+import lombok.extern.log4j.Log4j2;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.UUID;
+
+import static java.util.Optional.ofNullable;
+
+@Log4j2
+@Service
+public class MembershipsServiceImpl implements MembershipsService {
+
+    private final MembershipRepository membershipRepository;
+    private final RoleRepository roleRepository;
```

Comment:
I suggest to have the "roleRepository" field being of the "RolesService" type instead of the "RoleRepository" one. I prefer to have a service layer of domain X only accessing the repository layer of the same domain X as a rule. If an I/O operation of another domain is required, it should be done by the service layer of this domain and not by its repository layer. The reason for this is that we might miss an important business rule, which should be done in the service layer, by calling methods directly from the repository layer.